DOC Update changelog with list interface changes (#605)
silverstripe/developer-docs: en/08_Changelogs/6.0.0.md

If you reference any of these classes in your project or module, most likely in config if you have [`tractorcow/silverstripe-fluent`](https://github.com/tractorcow/silverstripe-fluent) installed, then you will need to update the references to the new class names.

### List interface changes {#list-interface-changes}

The [`SS_List`](api:SilverStripe\Model\List\SS_List) interface now includes the methods from the `Filterable`, `Limitable`, and `Sortable` interfaces, which have now been removed. This means that any class that implements `SS_List` must now also implement the methods from those interfaces.

> Many of the methods on `SS_List` are now strongly typed, along with the classes that implement it. This means that you will need to ensure that any custom classes that implement `SS_List` have the correct types for the methods that they implement.

Reviewer: ```suggestion
Many of the methods on `SS_List` and the classes that implement it are now strongly typed. This means that you will need to ensure that any custom classes that implement `SS_List` have the correct types for the methods that they implement.
```

Author: Updated
